### PR TITLE
Add tts notebook

### PR DESCRIPTION
**Pull request by blisc**
Updates notebook numbering
Add new notebook to showcase how duration and pitch control work in fastpitch

### CODE REVIEW

File: nemo/collections/tts/helpers/helpers.py [hunk 1]
```diff
@@ -372,3 +372,37 @@ def eval_tts_scores(
     ## fs was set 16,000, as pesq lib doesnt currently support felxible fs.
 
     return {'STOI': stoi_score, 'PESQ': pesq_score}
+
+
+def regulate_len(durations, enc_out, pace=1.0, mel_max_len=None):
```
Comment (redoctopus) [inline nemo/collections/tts/helpers/helpers.py:377]: Do we need the FastPitch license added to the header of this file w/ this function moved?

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+    "# !pip install wget unidecode\n",
+    "# BRANCH = 'main'\n",
+    "# !python -m pip install git+https://github.com/NVIDIA/NeMo.git@$BRANCH#egg=nemo_toolkit[tts]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Imports\n",
```
Comment (redoctopus) [inline tutorials/tts/2_Inference_DurationPitchControl.ipynb:60]: "Imports" -> "Setup" + update description? Since there are some helper functions and model downloads as well.

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+    "        audio = hifigan.convert_spectrogram_to_audio(spec=spec).to('cpu').numpy()\n",
+    "    return spec, audio, durs_pred, pitch_pred\n",
+    "\n",
+    "# Define a helper function to plot spectrograms with pitch and display the audio\n",
+    "def display_pitch(audio, pitch, sr=22050, durs=None):\n",
+    "    fig, ax = plt.subplots(figsize=(12, 6))\n",
+    "    spec = np.abs(librosa.stft(audio[0], n_fft=1024))\n",
+    "    # Check to see if pitch has been unnormalized\n",
+    "    if torch.abs(torch.mean(pitch)) <= 1.0:\n",
+    "        pitch = pitch * 65.72037058703644 + 214.72202032404294\n",
```
Comment (redoctopus) [inline tutorials/tts/2_Inference_DurationPitchControl.ipynb:108]: Explanation for magic numbers

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+    "    plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Duration Control\n",
+    "\n",
+    "This section is applicable to models that use a duration predictor module. This module is called the Length Regulator and was introduce in FastSpeech [1]. A list of NeMo models that support duration predictors are as follows:\n",
```
Comment (redoctopus) [inline tutorials/tts/2_Inference_DurationPitchControl.ipynb:125]: Minor typo
s/was introduce/was introduced

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+    "_, audio, *_ = str_to_audio(input_string)\n",
+    "print(f\"This is fastpitch speaking at the regular pace of 1.0. This example is {len(audio[0])/sr:.3f} seconds long.\")\n",
+    "ipd.display(ipd.Audio(audio, rate=sr))\n",
+    "\n",
+    "# We can speed up the speech by adjusting the pace\n",
+    "_, audio, *_ = str_to_audio(input_string, pace=1.2)\n",
+    "print(f\"This is fastpitch speaking at the faster pace of 1.2. This example is {len(audio[0])/sr:.3f} seconds long.\")\n",
+    "ipd.display(ipd.Audio(audio, rate=sr))\n",
+    "\n",
+    "# We can slow down the speech  by adjusting the pace\n",
```
Comment (redoctopus) [inline tutorials/tts/2_Inference_DurationPitchControl.ipynb:169]: Minor typo: double space

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb [hunk 6]
```diff
@@ -0,0 +1,490 @@
...
+    "```python\n",
+    "# Predict pitch\n",
+    "pitch_predicted = self.pitch_predictor(enc_out, enc_mask)  # Predicts a pitch that is normalized with speaker statistics \n",
+    "pitch_emb = self.pitch_emb(pitch.unsqueeze(1))  # A simple 1D convolution to map the float pitch to a embedding pitch\n",
+    "\n",
+    "enc_out = enc_out + pitch_emb.transpose(1, 2)  # We add the pitch to the encoder output\n",
+    "spec, *_ = self.decoder(input=len_regulated, seq_lens=dec_lens)  # We send the sum to the decoder to get the spectrogram\n",
+    "```\n",
+    "\n",
+    "Let's see the `pitch_predicted` for a sample text. You can run the below cell. You should get an image that looks like for the input `Hey, what is my pitch?`:\n",
```
Comment (redoctopus) [inline tutorials/tts/2_Inference_DurationPitchControl.ipynb:202]: Typo?
s/looks like for/looks like the following for

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb [hunk 7]
```diff
@@ -0,0 +1,490 @@
...
+    "display_pitch(audio_flat, pitch_flat, durs=durs_flat_pred)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Third we look at pitch flattening. To invert the pitch, we just multiply it by -1.\n",
```
Comment (redoctopus) [inline tutorials/tts/2_Inference_DurationPitchControl.ipynb:329]: s/pitch flattening/pitch inversion

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb [hunk 8]
```diff
@@ -0,0 +1,490 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# First we look at pitch shift. To shift the pitch up or down by some Hz, we can just add or subtract as needed\n",
+    "# Let's shift the pitch down by 50 Hz\n",
+    "# First, let's run the previous example and then shift down\n",
```
Comment (redoctopus) [inline tutorials/tts/2_Inference_DurationPitchControl.ipynb:266]: Can you pull these numbered comments out of the code cells and into text cells? Might be stylistically more consistent for a notebook

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb [hunk 9]
```diff
@@ -0,0 +1,490 @@
...
+    "display_pitch(audio_amp, pitch_amp, durs=durs_amp_pred)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Putting it all together\n",
+    "\n",
+    "Now that we understand how to control the duration and pitch of TTS models. We can show how to adjust the voice to sound more solemn (slower speed + lower pitch), or more excited (higher speed + higher pitch)."
```
Comment (redoctopus) [inline tutorials/tts/2_Inference_DurationPitchControl.ipynb:385]: Minor typo:
s/models. We/models, we

Comment (blisc) [reply tutorials/tts/2_Inference_DurationPitchControl.ipynb]: done

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb (hunk 8 above)
Comment (blisc) [reply tutorials/tts/2_Inference_DurationPitchControl.ipynb]: done

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb (hunk 7 above)
Comment (blisc) [reply tutorials/tts/2_Inference_DurationPitchControl.ipynb]: done

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb (hunk 6 above)
Comment (blisc) [reply tutorials/tts/2_Inference_DurationPitchControl.ipynb]: done

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb (hunk 5 above)
Comment (blisc) [reply tutorials/tts/2_Inference_DurationPitchControl.ipynb]: done

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb (hunk 4 above)
Comment (blisc) [reply tutorials/tts/2_Inference_DurationPitchControl.ipynb]: done

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb (hunk 3 above)
Comment (blisc) [reply tutorials/tts/2_Inference_DurationPitchControl.ipynb:109]: done

File: tutorials/tts/2_Inference_DurationPitchControl.ipynb (hunk 2 above)
Comment (blisc) [reply tutorials/tts/2_Inference_DurationPitchControl.ipynb]: done

File: nemo/collections/tts/helpers/helpers.py (hunk 1 above)
Comment (blisc) [reply nemo/collections/tts/helpers/helpers.py:407]: done

**Review by redoctopus — APPROVED** [review]
LGTM
